js/src/components/paid-ads/asset-group/texts-editor.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import GridiconCrossSmall from 'gridicons/dist/cross-small';
+
+/**
+ * Internal dependencies
+ */
+import AppButton from '.~/components/app-button';
+import AppInputControl from '.~/components/app-input-control';
+import AddAssetItemButton from './add-asset-item-button';
+import './texts-editor.scss';
+
+/**
+ * Renders a list of text inputs for managing the single type of asset texts.
+ *
+ * @param {Object} props React props.
+ * @param {string[]} [props.initialTexts=[]] Initial texts.
+ * @param {number} [props.minNumberOfTexts=0] Minimum number of texts.
+ * @param {number} [props.maxNumberOfTexts=0] Maximum number of texts.
+ * @param {number|number[]} props.maxCharacterCounts Maximum number of characters for each text. If the limits are the same, a single number can be used instead of an array.
+ * @param {string} props.addButtonText Text for the button to add a new text input.
+ * @param {string} [props.placeholder] Placeholder text.
+ * @param {Function} props.onChange Callback function to be called when the texts are changed.
+ */
+export default function TextsEditor( {
+	initialTexts = [],
+	minNumberOfTexts = 0,
+	maxNumberOfTexts = 0,
+	maxCharacterCounts,
+	addButtonText,
+	placeholder,
+	onChange,
+} ) {
+	const [ texts, setTexts ] = useState( () => {
+		const shortage = minNumberOfTexts - initialTexts.length;
```

Comment:
⚠️   We are not handling here the `maxNumberOfTexts` constraint. 

Hence if we add a component like this;:

```jsx
<TextsEditor
							minNumberOfTexts={ 3 }
							maxNumberOfTexts={ 3 }
							maxCharacterCounts={ [ 15, 30, 30, 30, 30 ] }
							placeholder={ __(
								'Headline',
								'google-listings-and-ads'
							) }
							addButtonText={ __(
								'Add headline',
								'google-listings-and-ads'
							) }
							initialTexts={ [ 'Hello', 'World', 'kkk', 'ppp' ] }
							onChange={ ( v ) =>
								// eslint-disable-next-line no-console
								console.log( 'Headline:onChange', v )
							}
						/>
```

It will render the 5 values even tho the maximum is 3

<img width="543" alt="Screenshot 2023-01-19 at 10 51 14" src="https://user-images.githubusercontent.com/5908855/213351341-58dd706d-c80f-4193-abf3-87b2f066cee1.png">

Reply:
Fixed in https://github.com/woocommerce/google-listings-and-ads/commit/0fe5fc9bc716542590ac57d3f06c7c9dce6b9470.

Reply:
Pushed another related adjustment https://github.com/woocommerce/google-listings-and-ads/pull/1854/commits/ba8b8910905d30494020fda6028732c2868313a7 to avoid a warning:
<img width="648" alt="image" src="https://user-images.githubusercontent.com/17420811/213383661-3dd4b063-6388-4981-8f32-ecf6ba9d01ce.png">